### PR TITLE
[makefile] Support for sed on Darwin

### PR DESCRIPTION
**Pull request by vladikkuzn**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Uses gnu version of sed while using Darwin OS
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Review by trasc — COMMENTED** [review]
/lgtm

File: Makefile [hunk 1]
```diff
@@ -307,7 +316,7 @@ artifacts: kustomize yq helm
 
 .PHONY: prepare-release-branch
 prepare-release-branch: yq kustomize
-	sed -r 's/v[0-9]+\.[0-9]+\.[0-9]+/$(RELEASE_VERSION)/g' -i README.md -i site/hugo.toml
+	$(SED) -r 's/v[0-9]+\.[0-9]+\.[0-9]+/$(RELEASE_VERSION)/g' -i README.md -i site/hugo.toml
```
Comment (alculquicondor) [inline Makefile:319]: I support the change of this line.

File: Makefile [hunk 2]
```diff
@@ -84,7 +84,16 @@ SHELL = /usr/bin/env bash -o pipefail
 
 # Setting SED allows macos users to install GNU sed and use the latter
 # instead of the default BSD sed.
-SED ?= /usr/bin/sed
+SED := /usr/bin/sed
```
Comment (alculquicondor) [inline Makefile:87]: I don't like having platform specific code.

Maybe we can have a simpler approach?

(in pseudocode)

```
SED=$(which gsed)
if SED is empty
  SED=$(which sed)
```

Comment (tenzen-y) [reply Makefile:87]: +1
Also, using `command -v` would be better for following GNU specifications.

Comment (tenzen-y) [reply Makefile:87]: I don't have any strong objections to using the `$SED --version`.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**